c/datasetjson.c

```diff
@@ -2029,7 +2032,32 @@ void respondWithDatasetMetadata(HttpResponse *response) {
 
   int fieldCount = defaultCSIFieldCount;
   char **csiFields = defaultCSIFields;
-
+  int addQualifiers = !strcmp(addQualifiersArg, "true");
+#define DSN_MAX_LEN 44
+  char dsnNameNullTerm[DSN_MAX_LEN] = {0};
+  memcpy(dsnNameNullTerm, dsnName.value, sizeof(dsnName.value));
+  nullTerminate(dsnNameNullTerm, sizeof(dsnNameNullTerm) - 1);
+  if (addQualifiers && dsnLen < DSN_MAX_LEN) {
+    int asteriskPos = lastIndexOf(dsnNameNullTerm, dsnLen,'*');
+    int dblAsteriskPos = indexOfString(dsnNameNullTerm, dsnLen, "**", 0); //"**" is only valid at the end of a query, cannot appear in the middle of a search
+    int periodPos = lastIndexOf(dsnNameNullTerm, dsnLen, '.');
+    char newDsn[DSN_MAX_LEN];
+    if (asteriskPos < 0 && dblAsteriskPos < 0) {
+      if(periodPos < 0 || periodPos != dsnLen - 1){ //-1 for null terminator.  Query in form of hlq1.hlq2
```

Comment:
Are you checking if the period is the last character? Then the comment is incorrect.

Reply:
This is checking that the period is NOT the last character.

Reply:
Okay, but the comment is still incorrect, isn't it? `dsnLen - 1` is because of 0 based indexing not null terminator.

Reply:
Yes that would be correct, i have amended it.